### ISSUE TITLE
Pass kwargs from WebMapContainer.get_service() to owslib.wms.WebMapService()

### ISSUE THREAD

**Issue by Joshdpaul** [raphaelquast/EOmaps#200]
When trying to add a WMS service to a Map, the current `WebMapContainer.get_service()` [function](https://eomaps.readthedocs.io/en/latest/generated/eomaps.webmap_containers.WebMapContainer.get_service.html) essentially passes a WMS URL to `owslib.wms.WebMapService()` but does not accept any kwargs for that function. I'd like to be able to pass kwargs here, especially `version='#.#.#'`.

Why this feature? The `owslib` function is usually able to parse the WMS version from XML GetCapabilities documents, but there are some instances where the version needs to be specified explicitly (otherwise, the WMS version defaults to 1.1.1 in the `owslib.wms.WebMapService()` [function](https://github.com/geopython/OWSLib/blob/5d6df7ccc7a83fa09054058e8e357892a5b3b770/owslib/wms.py#L22)). 

I run into this problem using my organization's Rasdaman WMS service. (See code example below.) Without specifying `version=1.3.0` to `owslib.wms.WebMapService()`, the function fails. This means I cannot add the WMS service using the `eomaps` `WebMapContainer.get_service()` function. I really like the simplicity of the `eomaps` WMS functions and would love to be able to use Rasdaman services! 

```
>>> from owslib.wms import WebMapService
>>> rasdaman_url = "https://apollo.snap.uaf.edu/rasdaman/ows?version=1.3.0&service=WMS&request=GetCapabilities" 
>>> #works with WMS version 1.3.0 specified
>>> r = WebMapService(rasdaman_url, version='1.3.0')
>>> list(r.contents)[0:3]
['alfresco_relative_flammability_30yr', 'alfresco_vegetation_mode_statistic', 'alfresco_vegetation_type_percentage']
>>> 
>>> #but does not work without WMS version 1.3.0 specified
>>> #OWSLib unable to parse version from XML, defaults to WMS 1.1.1
>>> r = WebMapService(rasdaman_url)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/UA/jdpaul3/mambaforge/envs/eomaps/lib/python3.11/site-packages/owslib/wms.py", line 50, in WebMapService
    return wms111.WebMapService_1_1_1(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/UA/jdpaul3/mambaforge/envs/eomaps/lib/python3.11/site-packages/owslib/map/wms111.py", line 74, in __init__
    self._capabilities = reader.read(self.url, timeout=self.timeout)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/UA/jdpaul3/mambaforge/envs/eomaps/lib/python3.11/site-packages/owslib/map/common.py", line 65, in read
    u = openURL(spliturl[0], spliturl[1], method='Get',
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/UA/jdpaul3/mambaforge/envs/eomaps/lib/python3.11/site-packages/owslib/util.py", line 212, in openURL
    raise ServiceException(req.text)
owslib.util.ServiceException: <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<ows:ExceptionReport version="2.0.1"
    xsi:schemaLocation="http://www.opengis.net/ows/2.0 http://schemas.opengis.net/ows/2.0/owsExceptionReport.xsd"
    xmlns:ows="http://www.opengis.net/ows/2.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xlink="http://www.w3.org/1999/xlink">
    <ows:Exception exceptionCode="InvalidParameterValue" locator="request">
        <ows:ExceptionText>Operation request contains an invalid parameter value</ows:ExceptionText>
    </ows:Exception>
</ows:ExceptionReport>
```

**Comment by raphaelquast** [comment]
Hey, thanks again for reporting!
 (and nice to hear that you like the simplicity ... that's one of my main goals with eomaps in general! )

I fully agree that the current implementation for custom webmap services has a lot of room for improvement...
(also the underlying cartopy implementations for webmap services are suboptimal and have some known bugs)

> Any PR that improves handling of webmap services is highly welcome!

---

Now to the good part:  
Since I was already working on EOmaps I gave it a quick shot and the good news is that with some minor
adjustments your wms-url works as expected! 🥳 

You can see the required changes in this PR:  https://github.com/raphaelquast/EOmaps/pull/201  (if all tests are OK I'll include it in the upcoming release!)


However, this is not yet enough to get your wms working since cartopy will start complaining that it does not know what epsg=3338 means... 
After explicitly telling cartopy how to handle the projection all works nicely!

So in short... If you run off the [wms_kwargs](https://github.com/raphaelquast/EOmaps/tree/wms_kwargs) branch, this works:


```python
# tell cartopy how to deal with the wms projection
from cartopy.io.ogc_clients import _CRS_TO_OGC_SRS
_CRS_TO_OGC_SRS[Maps.CRS.epsg(3338)] = ["EPSG:3338"]

from eomaps import Maps

m = Maps(3338)
m.add_feature.preset("coastline", "ocean")
wms = m.add_wms.get_service(
    "https://apollo.snap.uaf.edu/rasdaman/ows?version=1.3.0&service=WMS&request=GetCapabilities",
    version='1.3.0'
    )

w = wms.add_layer.annual_mean_temp()
```


<img src=https://github.com/raphaelquast/EOmaps/assets/22773387/bc43a913-0696-4bfc-93dc-0347d2ea56f8 width=50%/>

**Comment by raphaelquast** [comment]
@Joshdpaul I'm happy to report that this will be part of the next EOmaps version (to be released soon)

**Comment by raphaelquast** [comment]
[EOmaps v7.3.2](https://github.com/raphaelquast/EOmaps/releases/tag/v7.3.2) is released! Now you can pass kwargs to `owslib` when using custom webmap services!

**Comment by Joshdpaul** [comment]
Excellent! Thanks for adding this feature!
